test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py

```diff
@@ -390,6 +391,79 @@ def test_smmodelparallel_mnist_multigpu_multinode_efa(ecr_image, efa_instance_ty
         job_name=utils.unique_name_from_base('test-pt-smdmp-multinode-efa')
         invoke_pytorch_estimator(ecr_image, sagemaker_regions, estimator_parameter, job_name=job_name)
 
+@pytest.mark.integration("smmodelparallel")
+@pytest.mark.model("gpt2")
+@pytest.mark.processor("gpu")
+@pytest.mark.multinode(2)
+@pytest.mark.skip_cpu
+@pytest.mark.skip_py2_containers
+@pytest.mark.parametrize("test_script, num_processes", [("train_gpt_simple.py", 8)])
+@pytest.mark.efa()
+def test_smmodelparallel_gpt2_sdp_multinode_efa(ecr_image, efa_instance_type, sagemaker_regions, test_script, num_processes):
+    """
+    Tests pt gpt2 command via script mode
+    """
+    framework, framework_version = get_framework_and_version_from_tag(ecr_image)
+    if framework == "pytorch" and Version(framework_version) in SpecifierSet("<1.12.0"):
+        pytest.skip("Skipping the test for PT version before 1.12")
+    smp_version = 111
```

Comment:
does this needs to be hardcoded? Would it be the same for new versions for pytorch?

Reply:
This is intended to maintain the backward compatibility of SMP APIs, for the new PT versions, it could be same unless we introduce new APIs to SMP scripts.